src/components/MetadataPanel/index.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useEffect, useMemo } from "react";
+import { Divider } from "antd";
+import classNames from "classnames";
+
+import { AgentMetadata, MetadataDisplay } from "../../constants/interfaces";
+import { formatFloatForDisplay } from "../../util";
+import { FilledCaret } from "../Icons";
+import { AgentDisplayNode } from "../AgentTree";
+
+import styles from "./style.css";
+
+interface MetadataPanelProps {
+    selectedAgent: AgentMetadata;
+    uiDisplayData: AgentDisplayNode[];
+}
+
+const MetadataPanel: React.FC<MetadataPanelProps> = ({
+    selectedAgent,
+    uiDisplayData,
+}) => {
+    const [panelExpanded, setPanelExpanded] = React.useState(false);
+
+    const agentSelected = !!selectedAgent.uniqueId;
+    const selectedAgentNotRendered =
+        selectedAgent.agentType >= uiDisplayData.length;
+
+    useEffect(() => {
+        if (!agentSelected) {
+            setPanelExpanded(false);
+        } else {
+            setPanelExpanded(true);
+        }
+    }, [selectedAgent.uniqueId]);
```

Comment:
This effect is a little hard for me to follow. I think it would be clearer to just have 
`if (!selectedAgent) { setPanelExpanded(false)}`
or `if(!selectedAgent.uniqueId) { setPanelExpanded(false)}`
I don't think you need the constant `agentSelected`?

Reply:
I can remove the constant, I thought having `agentSelected` was more semantic/readable than `selectedAgent.uniqueId`. Either will work. Could also make it even more explicit, `agentIsSelected` would that be more clear?

Can't use `!selectedAgent` because its always truthy in this current version of the code.

Reply:
Deleted the constant and I'm using `selectedAgentMetadata.uniqueId` in its place per your feedback! Hopefully this is more readable to you!